bndtools.core/src/bndtools/editor/project/RunRequirementsPart.java

```diff
@@ -257,8 +259,8 @@ private void doAddBundle() {
             WizardDialog dialog = new WizardDialog(getSection().getShell(), wizard);
 
             if (Window.OK == dialog.open()) {
-                List<VersionedClause> result = wizard.getSelectedBundles();
-                List<Requirement> adding = new ArrayList<Requirement>(result.size());
+                final List<VersionedClause> result = wizard.getSelectedBundles();
+                final List<Requirement> adding = new ArrayList<Requirement>(result.size());
```

Comment:
Couldn't this be much simpler if `adding` was a Set? Then duplicate adds would just be ignored.